runtime-sdk/modules/evm/src/types.rs

```diff
@@ -222,6 +222,68 @@ mod eth {
         }
     }
 
-    impl_upstream_conversions!(H160, H256, U256);
+    impl_upstream_conversions_primitive_types!(H160, H256, U256);
+
+    /// Add solabi::Decode implementation for H160.
+    impl solabi::decode::Decode for H160 {
+        fn is_dynamic() -> bool {
+            false // Addresses are fixed-size
+        }
+
+        fn decode(
+            decoder: &mut solabi::decode::Decoder,
+        ) -> Result<Self, solabi::decode::DecodeError> {
+            // Decode as solabi::Address, then convert to H160
+            let solabi_addr = solabi::Address::decode(decoder)?;
+            Ok(H160::from(solabi_addr))
+        }
+    }
+
+    // Add solabi::Encode implementation for H160
+    impl solabi::encode::Encode for H160 {
+        fn size(&self) -> solabi::encode::Size {
+            // Delegate to solabi::Address's size implementation
+            solabi::Address::from(*self).size()
+        }
+
+        fn encode(&self, encoder: &mut solabi::encode::Encoder) {
+            // Convert to solabi::Address and encode
+            solabi::Address::from(*self).encode(encoder);
+        }
+    }
+
+    impl From<solabi::Address> for H160 {
+        fn from(addr: solabi::Address) -> Self {
+            H160(addr.0)
+        }
+    }
+
+    impl From<H160> for solabi::Address {
+        fn from(addr: H160) -> Self {
+            solabi::Address(addr.0)
+        }
+    }
+
+    impl From<solabi::U256> for U256 {
+        fn from(u: solabi::U256) -> Self {
+            let u64_array: [u64; 4] = [
+                (u.0[0] >> 64) as u64, // high 64 bits of first u128
```

Comment:
```suggestion
                (u.0[0] >> 64) as u64, // High 64 bits of first u128.
```
And similarly for the other three comments.